src/confed_json_generator.cpp

```diff
@@ -269,37 +318,19 @@ namespace
 
     Json::Value MakeDeviceForConfed(const Json::Value& config, TTemplateMap& deviceTemplates)
     {
-        auto dt = config["device_type"].asString();
-
         PDeviceTemplate deviceTemplate;
         try {
-            deviceTemplate = deviceTemplates.GetTemplate(dt);
+            deviceTemplate = deviceTemplates.GetTemplate(config["device_type"].asString());
         } catch (...) {
             return config;
         }
 
-        if (!deviceTemplate->WithSubdevices()) {
-            return MakeDeviceWithGroupsForConfed(config, deviceTemplate->GetTemplate());
-        }
-
-        Json::Value schema(deviceTemplate->GetTemplate());
         Json::Value newDev(config);
         ConvertPollIntervalToReadRateLimit(newDev);
-        JoinChannelsToGroups(newDev, schema);
-
-        TransformGroupsToSubdevices(schema, schema["subdevices"]);
-
-        Json::Value customChannels;
-        Json::Value standardChannels;
-        std::tie(standardChannels, customChannels) = SplitChannels(newDev, schema);
-        newDev.removeMember("channels");
-        if (!customChannels.empty()) {
-            newDev["channels"] = customChannels;
-        }
-        if (!standardChannels.empty()) {
-            TSubDevicesTemplateMap subDeviceTemplates(deviceTemplate->Type, schema);
-            MakeSubDevicesForConfed(standardChannels, subDeviceTemplates);
-            newDev["standard_channels"] = standardChannels;
+        if (!deviceTemplate->WithSubdevices()) {
```

Comment:
```suggestion
        if (deviceTemplate->WithSubdevices()) {
```
И дальше наоборот. Читать проще

Reply:
fixed